x/ecocredit/client/testsuite/grpc.go

```diff
@@ -434,106 +350,49 @@ func (s *IntegrationTestSuite) TestGetSellOrders() {
 			resp, err := rest.GetRequest(tc.url)
 			require.NoError(err)
 
-			var sellOrders ecocredit.QuerySellOrdersResponse
-			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &sellOrders)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
-			} else {
-				require.NoError(err)
-				require.NotNil(sellOrders.SellOrders)
-				require.Len(sellOrders.SellOrders, tc.expItems)
-			}
-		})
-	}
-}
-
-func (s *IntegrationTestSuite) TestGetSellOrdersByBatchDenom() {
-	val := s.network.Validators[0]
-	batchDenom := s.batchInfo.BatchDenom
-
-	testCases := []struct {
-		name     string
-		url      string
-		expErr   bool
-		errMsg   string
-		expItems int
-	}{
-		{
-			"invalid denom",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/batch-denom/%s", val.APIAddress, "foo"),
-			true,
-			"invalid denom",
-			0,
-		},
-		{
-			"valid request",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/batch-denom/%s", val.APIAddress, batchDenom),
-			false,
-			"",
-			3,
-		},
-		{
-			"valid request pagination",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/sell-orders/batch-denom/%s?pagination.limit=2", val.APIAddress, batchDenom),
-			false,
-			"",
-			2,
-		},
-	}
-
-	require := s.Require()
-	for _, tc := range testCases {
-		tc := tc
-		s.Run(tc.name, func() {
-			resp, err := rest.GetRequest(tc.url)
+			var res marketplace.QuerySellOrdersResponse
+			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &res)
 			require.NoError(err)
-
-			var sellOrders ecocredit.QuerySellOrdersByBatchDenomResponse
-			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &sellOrders)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
+			require.NotNil(res.SellOrders)
+			if tc.expItems > 0 {
+				require.Len(res.SellOrders, tc.expItems)
 			} else {
-				require.NoError(err)
-				require.NotNil(sellOrders.SellOrders)
-				require.Len(sellOrders.SellOrders, tc.expItems)
+				require.Greater(len(res.SellOrders), 0)
 			}
```

Comment:
If `expItems` is not greater than `0` than require greater than `0`...? We should not be using `-1` for the expected number of items.

Reply:
see above comment